src/app/core/export/download-service/download.service.spec.ts

```diff
@@ -77,11 +77,12 @@ describe("DownloadService", () => {
       '"_id","_rev","propOne","propTwo"' +
       DownloadService.SEPARATOR_ROW +
       '"TestForCsvEntity:1","2","first","second"';
+    spyOn(service, "exportFile").and.returnValue(expected);
     const result = await service.createCsv([test]);
     expect(result).toEqual(expected);
   });
 
-  it("should transform object properties to their label for export", async () => {
+  it("should transform schema label to their label for export when label is defined in entity schema", async () => {
```

Comment:
although the test name says it, I do not see and "expect" check regarding the column headers of the exported csv (which would be the values in `row[0]`)

This test originally checked whether the value of a `ConfigurableEnumValue` (which is a dropdown options field) gets exported by its human-readable label rather than the option id. That is still an important test, and is still tested here.
But your new functionality is missing a real test case. Maybe it would be better to create another, new test for your part.

Reply:
@sleidig I have added a new test case for check labels in CSV export

Reply:
thanks! This covers all the logic well now :+1: 

Just a few glitches in your test code - I am pushing a small commit to fix that.